### DISCUSSION TITLE
How to access the queryClient outside the provider? e.g. to invalidate queries in Cypress tests

### DISCUSSION THREAD

**Discussion by mrmartineau** [TanStack/query#2727] [Q&A] [answered]
Is it possible to access my app's `queryClient` outside React?

In my some of my Cypress tests I fetch or mutate some data but then I'd like to invalidate all/part of the `queryClient` cache afterwards.

Is that possible? 

I have tried importing the "same" `queryClient` that is used in the app, but it doesn't work.

ℹ️  I include these fetch/mutations in my tests merely to allow me to bypass convoluted steps that user's would normally take in the app which already have Cypress tests.

**Comment by TkDodo** [comment, marked as answer]
https://stackoverflow.com/questions/69344045/how-to-access-the-react-query-queryclient-outside-the-provider-e-g-to-invali
